src/ModelIO/MDLAsset.cs

```diff
@@ -1,12 +1,23 @@
 #if XAMCORE_2_0 || !MONOMAC
 using System;
+using XamCore.ObjCRuntime;
 namespace XamCore.ModelIO {
 	public partial class MDLAsset {
 		public MDLObject this [nuint index] {
 			get {
 				return GetObject (index);
 			}
 		}
+
+		[iOS (10,3), TV (10,2), Mac (10,12,4)]
+		public IMDLComponent this [Protocol key] {
```

Comment:
there's already a `this` that returns something else... check FxDG but I don't think we should expose it this way. That can get confusing source wise when the argument type is not clear.